target/linux/x86/base-files/lib/upgrade/platform.sh

```diff
@@ -39,19 +59,42 @@ platform_check_image() {
 platform_copy_config() {
 	local partdev parttype=ext4
 
-	if export_partdevice partdev 1; then
+	if (is_efi_system && export_partdevice_label partdev openwrt_boot) \
+		|| export_partdevice partdev 1 ; then
+		mkdir -p /tmp/boot
 		part_magic_fat "/dev/$partdev" && parttype=vfat
-		mount -t $parttype -o rw,noatime "/dev/$partdev" /mnt
-		cp -af "$UPGRADE_BACKUP" "/mnt/$BACKUP_FILE"
-		umount /mnt
+		mount -t $parttype -o rw,noatime "/dev/$partdev" /tmp/boot
+		cp -af "$UPGRADE_BACKUP" "/tmp/boot/$BACKUP_FILE"
+		umount /tmp/boot
 	fi
 }
 
 platform_do_bootloader_upgrade() {
 	local bootpart parttable=msdos
 	local diskdev="$1"
 
-	if export_partdevice bootpart 1; then
+	if is_efi_system; then
+		export_partdevice_label efipart "EFI system partition" || {
```

Comment:
Need to fix capitalisation here -
EFI System Partition

``
2147483648 bytes (2.1 GB, 2.0 GiB) copied, 17.9493 s, 120 MB/s
Cannot find 'EFI system partition'
Abort (y/N): n
Fri Aug 25 03:31:21 UTC 2023 upgrade: Upgrade completed
Reboot now? (y/N): y
``

and the blkid output is
``
/dev/sda1: SEC_TYPE="msdos" LABEL_FATBOOT="ESP" LABEL="ESP" UUID="64E6-B62C" BLOCK_SIZE="512" TYPE="vfat" PARTLABEL="EFI System Partition" PARTUUID="a53d301e-a488-bbcc-cf3e-d28647342b01"
``

Reply:
I did change it once following `gdisk` (or was it something else?) default as far as I remember.